Site/SiteAnalyticsModule.php

```diff
@@ -389,6 +584,186 @@ public function getFacebookPixelCommandsInlineJavascript()
 	}
 
 	// }}}
+	// {{{ protected function initFacebookPixelCommands()
+
+	protected function initFacebookPixelCommands()
+	{
+		// Default commands for all sites:
+		// * Track the page view.
+		$this->facebook_pixel_commands = array(
+			array(
+				'track',
+				'PageView',
+			),
+		);
+	}
+
+	// }}}
+	// {{{ protected function displayFacebookPixelImage()
+
+	protected function displayFacebookPixelImage()
+	{
+		if ($this->hasFacebookPixel()) {
+			$image = $this->getFacebookPixelImage();
+			if ($image != '') {
+				echo $image;
+			}
+		}
+	}
+
+	// }}}
+	// {{{ protected function displayFacebookPixel()
+
+	protected function displayFacebookPixel()
+	{
+		if ($this->hasFacebookPixel()) {
+			$js = $this->getFacebookPixelInlineJavascript();
+			if ($js != '') {
+				Swat::displayInlineJavaScript($js);
+			}
+		}
+	}
+
+	// }}}
+	// {{{ protected function getFacebookPixelCommand()
+
+	protected function getFacebookPixelCommand($command)
+	{
+		if (!is_array($command)) {
+			$command = array($command);
+		}
+
+		return sprintf(
+			'fbq(%s);',
+			implode(', ', array_map('json_encode', $command))
+		);
+	}
+
+	// }}}
+
+	// Twitter
+	// {{{ public function hasTwitterPixel()
+
+	public function hasTwitterPixel()
+	{
+		return (
+			$this->twitter_track_pixel_id != '' &&
+			!$this->analytics_opt_out
+		);
+	}
+
+	// }}}
+	// {{{ public function pushTwitterPixelCommands()
+
+	public function pushTwitterPixelCommands(array $commands)
+	{
+		foreach ($commands as $name => $value) {
+			$this->twitter_pixel_commands[$name] = $value;
```

Comment:
I'll remove the prepend method. Twitter works different from Facebook or GA in how the commands function.